packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/ClientContext.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { OntologyDefinition, ThinClient } from "@osdk/api";
+import type { OntologyObjectV2 } from "@osdk/gateway/types";
+import type { OntologyObject } from "../..";
+
+export interface ClientContext {
```

Comment:
This is intended to be the core of what you need to pass around in order to keep making network requests at any arbitrary depth of API calls.